src/pgwire/src/protocol.rs

```diff
@@ -403,6 +409,12 @@ where
             None => State::Done,
         };
 
+        let status = match next_state {
+            State::Ready | State::Done => "success",
+            State::Drain => "error",
```

Comment:
Any particular reason we want the client code to pass in a raw status string rather than using a constant or a dedicated function (`message_status_success.inc()` etc.)?

Reply:
it seemed plausible that we might want something more detailed than just "success" and "failure" at some point in the future in some of these cases

Reply:
I'm not sure that using `error` here makes sense, or that using `success` for `Done` is correct. For example, `Done` is produced when there was a fatal error that will cause the connection to close (https://github.com/MaterializeInc/materialize/blob/3c9876d6bbaa2da26283f2abb6e8f90bd52192d7/src/pgwire/src/protocol.rs#L1761) and `Drain` is used when a user tries to execute a statement in an aborted transaction (which can happen trivially and doesn't mean anything bad per se).

These are not messages users ever send or receive, but are internal state machine flags. I am not currently convinced they should be measured. And if they should, probably not like this.

Reply:
yeah, this was just restoring a metric that previously existed in #12869, but i'm not convinced it's actually meaningful at all, i'll probably just remove it